TestPionE2ESimplePSK is flaky

On CI environment, `TestPionE2ESimplePSK` sometimes fail due to timeout.

```
=== RUN   TestPionE2ESimplePSK
goroutine profile: total 4
1 @ 0x457eef 0x42db56 0x42db2c 0x42d8b5 0x6b9e57 0x6b9872 0x485a61
#	0x6b9e56	github.com/pion/dtls/v2/internal/udp.(*Listener).getConn+0x326	/home/travis/gopath/src/github.com/pion/dtls/internal/udp/conn.go:144
#	0x6b9871	github.com/pion/dtls/v2/internal/udp.(*Listener).readLoop+0x171	/home/travis/gopath/src/github.com/pion/dtls/internal/udp/conn.go:119

1 @ 0x457eef 0x42e84f 0x42e825 0x42e5db 0x53ae12 0x540c89 0x53a6c4 0x53ccf4 0x53b56c 0x7411a5 0x457adc 0x485a61
#	0x53ae11	testing.(*T).Run+0x6d1		/home/travis/.gimme/versions/go1.12.linux.amd64/src/testing/testing.go:917
#	0x540c88	testing.runTests.func1+0xa8	/home/travis/.gimme/versions/go1.12.linux.amd64/src/testing/testing.go:1157
#	0x53a6c3	testing.tRunner+0x163		/home/travis/.gimme/versions/go1.12.linux.amd64/src/testing/testing.go:865
#	0x53ccf3	testing.runTests+0x523		/home/travis/.gimme/versions/go1.12.linux.amd64/src/testing/testing.go:1155
#	0x53b56b	testing.(*M).Run+0x2eb		/home/travis/.gimme/versions/go1.12.linux.amd64/src/testing/testing.go:1072
#	0x7411a4	main.main+0x344			_testmain.go:250
#	0x457adb	runtime.main+0x20b		/home/travis/.gimme/versions/go1.12.linux.amd64/src/runtime/proc.go:200

1 @ 0x457eef 0x46922d 0x469203 0x468f9d 0x48ec19 0x48fe4a 0x6b8e98 0x71c417 0x73dfdd 0x73ac2c 0x73b2cd 0x53a6c4 0x485a61
#	0x468f9c	sync.runtime_SemacquireMutex+0x3c				/home/travis/.gimme/versions/go1.12.linux.amd64/src/runtime/sema.go:71
#	0x48ec18	sync.(*Mutex).Lock+0x148					/home/travis/.gimme/versions/go1.12.linux.amd64/src/sync/mutex.go:134
#	0x48fe49	sync.(*RWMutex).Lock+0x49					/home/travis/.gimme/versions/go1.12.linux.amd64/src/sync/rwmutex.go:93
#	0x6b8e97	github.com/pion/dtls/v2/internal/udp.(*Listener).Close+0x67	/home/travis/gopath/src/github.com/pion/dtls/internal/udp/conn.go:45
#	0x71c416	github.com/pion/dtls/v2.(*Listener).Close+0x76			/home/travis/gopath/src/github.com/pion/dtls/listener.go:46
#	0x73dfdc	github.com/pion/dtls/v2/e2e.assertE2ECommunication.func3+0x1cc	/home/travis/gopath/src/github.com/pion/dtls/e2e/e2e_test.go:143
#	0x73ac2b	github.com/pion/dtls/v2/e2e.assertE2ECommunication+0x3db	/home/travis/gopath/src/github.com/pion/dtls/e2e/e2e_test.go:177
#	0x73b2cc	github.com/pion/dtls/v2/e2e.TestPionE2ESimplePSK+0xfc		/home/travis/gopath/src/github.com/pion/dtls/e2e/e2e_test.go:235
#	0x53a6c3	testing.tRunner+0x163						/home/travis/.gimme/versions/go1.12.linux.amd64/src/testing/testing.go:865

1 @ 0x599d5e 0x599b2a 0x5959fc 0x738fd7 0x485a61
#	0x599d5d	runtime/pprof.writeRuntimeProfile+0x9d			/home/travis/.gimme/versions/go1.12.linux.amd64/src/runtime/pprof/pprof.go:708
#	0x599b29	runtime/pprof.writeGoroutine+0xc9			/home/travis/.gimme/versions/go1.12.linux.amd64/src/runtime/pprof/pprof.go:670
#	0x5959fb	runtime/pprof.(*Profile).WriteTo+0x4fb			/home/travis/.gimme/versions/go1.12.linux.amd64/src/runtime/pprof/pprof.go:329
#	0x738fd6	github.com/pion/transport/test.TimeOut.func1+0x96	/home/travis/gopath/pkg/mod/github.com/pion/transport@v0.8.10/test/util.go:18

panic: timeout

goroutine 129 [running]:
github.com/pion/transport/test.TimeOut.func1()
	/home/travis/gopath/pkg/mod/github.com/pion/transport@v0.8.10/test/util.go:21 +0x164
created by time.goFunc
	/home/travis/.gimme/versions/go1.12.linux.amd64/src/time/sleep.go:169 +0x52
FAIL	github.com/pion/dtls/v2/e2e	39.267s
```

Seems still flaky.
```
=== RUN   TestPionE2ESimplePSK/TLS_PSK_WITH_AES_128_CCM_8
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x8220833]

goroutine 947 [running]:
testing.tRunner.func1(0xa09e280)
	/usr/local/go/src/testing/testing.go:830 +0x345
panic(0x825d360, 0x84462a8)
	/usr/local/go/src/runtime/panic.go:522 +0x16e
github.com/pion/dtls/v2/e2e.(*comm).assert.func1(0xa012230, 0xa09e280)
	/go/src/github.com/pion/dtls/e2e/e2e_test.go:120 +0x93
runtime.Goexit()
	/usr/local/go/src/runtime/panic.go:406 +0xd6
testing.(*common).FailNow(0xa09e280)
	/usr/local/go/src/testing/testing.go:609 +0x2e
testing.(*common).Fatal(0xa09e280, 0xa4066c0, 0x1, 0x1)
	/usr/local/go/src/testing/testing.go:666 +0x62
github.com/pion/dtls/v2/e2e.(*comm).assert.func2(0xa012230, 0xa09e280)
	/go/src/github.com/pion/dtls/e2e/e2e_test.go:140 +0x193
github.com/pion/dtls/v2/e2e.(*comm).assert(0xa012230, 0xa09e280)
	/go/src/github.com/pion/dtls/e2e/e2e_test.go:163 +0x8d
github.com/pion/dtls/v2/e2e.testPionE2ESimplePSK.func1(0xa09e280)
	/go/src/github.com/pion/dtls/e2e/e2e_test.go:280 +0x2a7
testing.tRunner(0xa09e280, 0xa2aa900)
	/usr/local/go/src/testing/testing.go:865 +0x97
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:916 +0x2b5
FAIL	github.com/pion/dtls/v2/e2e	7.302s
?   	github.com/pion/dtls/v2/examples/dial	[no test files]
?   	github.com/pion/dtls/v2/examples/dial-psk	[no test files]
?   	github.com/pion/dtls/v2/examples/listen	[no test files]
?   	github.com/pion/dtls/v2/examples/listen-psk	[no test files]
?   	github.com/pion/dtls/v2/examples/util	[no test files]
?   	github.com/pion/dtls/v2/internal/closer	[no test files]
```
The panic seems caused by the problem in Go 1.12 testing package (in `FailNow()` func).
Anyway, it entered
```go
      case err := <-c.errChan:
        t.Fatal(err)
```

```
=== RUN   TestPionE2ESimplePSK
=== RUN   TestPionE2ESimplePSK/TLS_PSK_WITH_AES_128_CCM
=== RUN   TestPionE2ESimplePSK/TLS_PSK_WITH_AES_128_CCM_8
=== RUN   TestPionE2ESimplePSK/TLS_PSK_WITH_AES_128_GCM_SHA256
--- PASS: TestPionE2ESimplePSK (1.01s)
    e2e_test.go:262: Selected port 46350
    --- PASS: TestPionE2ESimplePSK/TLS_PSK_WITH_AES_128_CCM (0.20s)
    --- PASS: TestPionE2ESimplePSK/TLS_PSK_WITH_AES_128_CCM_8 (0.20s)
    --- PASS: TestPionE2ESimplePSK/TLS_PSK_WITH_AES_128_GCM_SHA256 (0.20s)
=== RUN   TestPionE2ESimple
=== RUN   TestPionE2ESimple/TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
=== RUN   TestPionE2ESimple/TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA
--- PASS: TestPionE2ESimple (0.82s)
    e2e_test.go:228: Selected port 53287
    --- PASS: TestPionE2ESimple/TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 (0.21s)
    --- PASS: TestPionE2ESimple/TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA (0.21s)
=== RUN   TestPionE2ESimplePSK
=== RUN   TestPionE2ESimplePSK/TLS_PSK_WITH_AES_128_CCM
=== RUN   TestPionE2ESimplePSK/TLS_PSK_WITH_AES_128_CCM_8
listen udp 127.0.0.1:49700: bind: address already in use
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
```
The e2e tests uses selected port among all test cases in each test. It may cause conflict of the port.

--- a/e2e/e2e_test.go
+++ b/e2e/e2e_test.go
@@ -220,8 +220,6 @@ func testPionE2ESimple(t *testing.T, server, client func(*comm)) {
 	report := test.CheckRoutines(t)
 	defer report()
 
-	serverPort := randomPort(t)
-
 	for _, cipherSuite := range []dtls.CipherSuiteID{
 		dtls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
 		dtls.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,
@@ -241,6 +239,7 @@ func testPionE2ESimple(t *testing.T, server, client func(*comm)) {
 				CipherSuites:       []dtls.CipherSuiteID{cipherSuite},
 				InsecureSkipVerify: true,
 			}
+			serverPort := randomPort(t)
 			comm := newComm(ctx, cfg, cfg, serverPort, server, client)
 			comm.assert(t)
 		})
@@ -253,8 +252,6 @@ func testPionE2ESimplePSK(t *testing.T, server, client func(*comm)) {
 
 	report := test.CheckRoutines(t)
 	defer report()
-
-	serverPort := randomPort(t)
 
 	for _, cipherSuite := range []dtls.CipherSuiteID{
 		dtls.TLS_PSK_WITH_AES_128_CCM,
@@ -273,6 +270,7 @@ func testPionE2ESimplePSK(t *testing.T, server, client func(*comm)) {
 				PSKIdentityHint: []byte{0x01, 0x02, 0x03, 0x04, 0x05},
 				CipherSuites:    []dtls.CipherSuiteID{cipherSuite},
 			}
+			serverPort := randomPort(t)
 			comm := newComm(ctx, cfg, cfg, serverPort, server, client)
 			comm.assert(t)
 		})
@@ -285,8 +283,6 @@ func testPionE2EMTUs(t *testing.T, server, client func(*comm)) {
 
 	report := test.CheckRoutines(t)
 	defer report()
-
-	serverPort := randomPort(t)
 
 	for _, mtu := range []int{
 		10000,
@@ -309,6 +305,7 @@ func testPionE2EMTUs(t *testing.T, server, client func(*comm)) {
 				InsecureSkipVerify: true,
 				MTU:                mtu,
 			}
+			serverPort := randomPort(t)
 			comm := newComm(ctx, cfg, cfg, serverPort, server, client)
 			comm.assert(t)
 		})

--- a/e2e/e2e_test.go
+++ b/e2e/e2e_test.go
@@ -112,19 +112,16 @@ func (c *comm) assert(t *testing.T) {
 	go c.server(c)
 
 	defer func() {
-		c.clientMutex.Lock()
-		c.serverMutex.Lock()
-		defer c.clientMutex.Unlock()
-		defer c.serverMutex.Unlock()
-
-		if err := c.clientConn.Close(); err != nil {
-			t.Fatal(err)
+		if c.clientConn != nil {
+			if err := c.clientConn.Close(); err != nil {
+				t.Fatal(err)
+			}
 		}
-
-		if err := c.serverConn.Close(); err != nil {
-			t.Fatal(err)
+		if c.serverConn != nil {
+			if err := c.serverConn.Close(); err != nil {
+				t.Fatal(err)
+			}
 		}
-
 		if c.serverListener != nil {
 			if err := c.serverListener.Close(); err != nil {
 				t.Fatal(err)

--- a/e2e/e2e_v113_test.go
+++ b/e2e/e2e_v113_test.go
@@ -26,8 +26,6 @@ func testPionE2ESimpleED25519(t *testing.T, server, client func(*comm)) {
 	report := test.CheckRoutines(t)
 	defer report()
 
-	serverPort := randomPort(t)
-
 	for _, cipherSuite := range []dtls.CipherSuiteID{
 		dtls.TLS_ECDHE_ECDSA_WITH_AES_128_CCM,
 		dtls.TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8,
@@ -53,6 +51,7 @@ func testPionE2ESimpleED25519(t *testing.T, server, client func(*comm)) {
 				CipherSuites:       []dtls.CipherSuiteID{cipherSuite},
 				InsecureSkipVerify: true,
 			}
+			serverPort := randomPort(t)
 			comm := newComm(ctx, cfg, cfg, serverPort, server, client)
 			comm.assert(t)
 		})

